### PR TITLE
hopefully made mineral doors less fucky

### PR DESCRIPTION
**Pull request by Spacedong**
seems to be an improvement from my testing, fixes issue #2938

### CODE REVIEW

**Comment by Scinexx** [comment]
thank god

**Comment by tortellinitony** [comment]
hopefully means you didn't fucking test it

**Comment by ArcLumin** [comment]
knowing dongcode this spawns 10 mobs when you open them

**Comment by Spacedong** [comment]
and the mobs are pizza mobs :^)

**Comment by KittyKathy** [comment]
Maybe you could also fix them being destroyed by 1 disabler shot?
Diamond door shot by a single disabler = no door anymore.

**Comment by Pyko1** [comment]
eh

should not that be something like?
```
/obj/structure/mineral_door/proc/Open()
	isSwitchingStates = 1
	playsound(loc, openSound, 100, 1)
	flick("[mineralType]opening",src)
	spawn(10)
		density = 0
		opacity = 0
		state = 1
		air_update_turf(1)
		update_icon()
		isSwitchingStates = 0

	if(close_delay != -1)
		spawn(close_delay)
			if(!isSwitchingStates && state == 1)
				Close()
```

**Comment by Pyko1** [comment]
I mean currently it has spawn(10) for what? nothing?

**Comment by SpankMasterson** [comment]
kk +1

**Comment by blobbernaut** [comment]
Yeah what pyko said is right

**Comment by blobbernaut** [comment]
The delay of 10 is for the door animation

File: code/game/objects/structures/mineral_doors.dm [hunk 1]
```diff
@@ -88,7 +88,7 @@
 	isSwitchingStates = 1
 	playsound(loc, openSound, 100, 1)
 	flick("[mineralType]opening",src)
-	sleep(10)
+	spawn(10)
```
Comment (JohnGinnane) [inline code/game/objects/structures/mineral_doors.dm:91]: You'll need to indent the code below so it does that code after 1 second

**Comment by ArcLumin** [comment]
fix your fucking code spacemong

**Comment by Pyko1** [comment]
more heads and code fixes needed

**Comment by ArcLumin** [comment]
It's a bugfix so no head input needed.

But fix your code, dong. Stop spurting out PRs, they're not sluts, you can't pump and dump.

**Comment by blobbernaut** [comment]
@Spacedong

**Comment by ArcLumin** [comment]
And minerals have nothing to do with doors as they are represented at Home Depot don't be a fucking faggot.

**Comment by SpankMasterson** [comment]
did this blow up the server or something god damn you spacedong **go fuck you you YOU!
**

**Pull request closed without merging**